pkg/reconciler/contour/config/contour.go

```diff
@@ -70,6 +75,40 @@ func NewContourFromConfigMap(configMap *corev1.ConfigMap) (*Contour, error) {
 		return nil, err
 	}
 
+	cors, ok := configMap.Data[corsPolicy]
+	if ok {
+		if err := yaml.Unmarshal([]byte(cors), &contourCORSPolicy); err != nil {
+			return nil, err
+		}
+
+		if len(contourCORSPolicy.AllowOrigin) == 0 || len(contourCORSPolicy.AllowMethods) == 0 {
+			return nil, fmt.Errorf("the following fields are required but are missing or empty: %s.allowOrigin and %s.allowMethods", corsPolicy, corsPolicy)
+		}
+
+		fields := [][]v1.CORSHeaderValue{
+			contourCORSPolicy.AllowMethods,
+			contourCORSPolicy.AllowHeaders,
+			contourCORSPolicy.ExposeHeaders,
+		}
+		for _, field := range fields {
+			if len(field) > 0 {
+				var validOption = regexp.MustCompile("^[a-zA-Z0-9!#$%&'*+.^_`|~-]+$")
+				for _, option := range field {
+					if !validOption.MatchString(string(option)) {
+						return nil, fmt.Errorf("%s.allowMethods, %s.allowHeaders, and %s.exposeHeaders have to be validly formatted. Option %s is invalid", corsPolicy, corsPolicy, corsPolicy, option)
+					}
+				}
+			}
+		}
+
+		if len(contourCORSPolicy.MaxAge) > 0 {
+			_, err := time.ParseDuration(contourCORSPolicy.MaxAge)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse %s.maxAge: %w", corsPolicy, err)
+			}
```

Comment:
Not really validating max age here - in theory we would allow negative numbers here which the contour API does not

https://github.com/projectcontour/contour/blob/fe259fa4f9060210b4e43e49aab9a5daedf84c94/apis/projectcontour/v1/httpproxy.go#L523C27-L526

Reply:
Good catch! used their regex.